### DISCUSSION TITLE
job notification

### DISCUSSION THREAD

**Discussion by alexkrush** [quartznet/quartznet#1075] [Q&A] [answered]
Hi,
I have a job in .net core 3.1 project that launches an async process in the database, waits till the database process is finished and proceed further with it's tasks. What will be the best way to implement the notification of my "running job instance" regarding the completion of the database process?
Many thanks in advance and rgds
Alexander

**Comment by lahma** [comment, marked as answer]
I think there's no silver bullet for this. If the process runs inside the database it's quite database dependent how you can raise notifications back to interested parties. One option would be run job in while loop sleeping and waiting that there's row in database corresponding to job's fire instance id (unique for each invocation), job would pass this to database logic.

If you decide to wait/sleep in job, you need to make sure you handle parallel invocations if the job takes longer than what it takes to start a new instance - if that would be problem (building backlog of jobs).

**Comment by asulwer** [comment]
jobs do not raise an event when they have completed?

**Reply by lahma** [reply]
Yes they do. This was a question about database notifying the job about its completion.
